examples_gbs/run_tutorial_max_clique.py

```diff
@@ -0,0 +1,158 @@
+
+##############################################################################
+# Let's take a look at some of these cliques. What are the clique sizes in the first ten samples?
+# What is the average clique size? How about the largest and smallest clique size?
+
+clique_sizes = [len(s) for s in shrunk]
+print('First ten clique sizes = ', clique_sizes[:10])
+print('Average clique size = {:.3f}'.format(np.mean(clique_sizes)))
+print('Maximum clique size = ', np.max(clique_sizes))
+print('Minimum clique size = ', np.min(clique_sizes))
+
+##############################################################################
+# Even in the first few samples, we've already identified larger cliques than the 4-node clique
+# we studied before. Awesome! Indeed, this simple shrinking strategy gives cliques with average
+# size of roughly five. We can enlarge these cliques by searching for larger cliques in their
+# vicinity. We'll do this by taking ten iterations of local search and studying the results.
+# Note: this may take a few seconds.
+
+searched = [gbs.clique.search(s, TA_graph, 10) for s in shrunk]
+clique_sizes = [len(s) for s in searched]
+print('First two cliques = ', searched[:2])
+print('Average clique size = {:.3f}'.format(np.mean(clique_sizes)))
+
+##############################################################################
+# Wow! Local search is very helpful 🤩. Looking at the cliques we've found, we learn that the largest
+# cliques in the graph have size eight, and there are many of them. Let's take a look at the first
+# one we found
+
+clique = gbs.plot.plot_graph(TA_graph, searched[0])
+plotly.offline.plot(clique, filename="maximum_clique.html")
+
+##############################################################################
+# .. raw:: html
+#     :file: ../../examples_gbs/maximum_clique.html
+
+##############################################################################
+# The TACE-AS graph is relatively small, so finding cliques is not particularly difficult. A
+# tougher challenge is the 300-node ``p_hat300-1`` random graph from the DIMACS maximum clique
+# dataset. In this section of the tutorial, we'll write a short program that uses GBS samples in
+# combination with local search to identify large cliques in this graph.
+#
+
+Phat = gbs.data.PHat()  # Load data
+phat_graph = nx.Graph(Phat.adj)  # Obtain graph
+phat_subgraphs = gbs.sample.to_subgraphs(phat_graph, Phat)  # Convert samples into subgraphs
+counts = Phat.counts()  # Post-select on samples
+samples = [phat_subgraphs[i] for i in range(len(Phat)) if counts[i] >= 16]
+
+shrunk = [gbs.clique.shrink(s, phat_graph) for s in samples]  # Shrink subgraphs to cliques
+searched = [gbs.clique.search(s, phat_graph, 10) for s in shrunk]  # Perform local search
+clique_sizes = [len(s) for s in searched]
+largest_clique = searched[np.argmax(clique_sizes)]  # Identify largest clique found
+print('Largest clique found is = ', largest_clique)

```

Comment:
I like the flow of looking at the TACE-AS graph then showing how to write a program for the PHat graph